src/shared/utilities/vsCodeUtils.ts

```diff
@@ -71,3 +71,10 @@ export async function activateExtension<T>(
 export function promisifyThenable<T>(thenable: Thenable<T>): Promise<T> {
     return new Promise((resolve, reject) => thenable.then(resolve, reject))
 }
+
+export function isUntitledScheme(path: string | vscode.Uri): boolean {
+    if (typeof path === 'string') {
+        return path.startsWith('untitled:')
+    }
+    return path.scheme === 'untitled'
+}
```

Comment:
Personally I wouldn't accept a string here as it would encourage opaque URIs:
```suggestion
export function isUntitledDocument(document: Pick<vscode.TextDocument, 'uri'>): boolean {
    return document.uri.scheme === 'untitled'
}
```